src/resolvers/categoryResolver.test.ts

```diff
@@ -1,2 +1,67 @@
-// TODO Write test cases
-// describe('categories() test cases', categoriesTestCases);
+import { assert } from 'chai';
+import {
+  graphqlUrl,
+  saveCategoryDirectlyToDb,
+  saveMainCategoryDirectlyToDb,
+} from '../../test/testUtils';
+import axios from 'axios';
+import {
+  getCategoryData,
+  getMainCategoriesData,
+} from '../../test/graphqlQueries';
+import { generateRandomString } from '../utils/utils';
+
+describe('mainCategoryTestCases() test cases', mainCategoryTestCases);
+describe('categoryTestCases() test cases', categoryTestCases);
+
+function mainCategoryTestCases() {
+  it('Should show main categories', async () => {
+    const title = generateRandomString(10);
+    await saveMainCategoryDirectlyToDb({
+      banner: '',
+      description: '',
+      slug: generateRandomString(10),
+      title,
+    });
+
+    const categoryResponse = await axios.post(graphqlUrl, {
+      query: getMainCategoriesData,
+      variables: {},
+    });
+    const result = categoryResponse.data.data.mainCategories;
+    assert.isNotEmpty(result);
+
+    assert.equal(result[result.length - 1].title, title);
+  });
+}
+
+function categoryTestCases() {
+  it('Should show  categories', async () => {
+    const name = generateRandomString(10);
+    const mainCategory = await saveMainCategoryDirectlyToDb({
+      banner: '',
+      description: '',
+      slug: 'environment-and-energy-test2',
+      title: 'Economics & Infrastructure test2',
+    });
+
+    await saveCategoryDirectlyToDb({
+      name,
+      mainCategory,
+      value: 'Agriculture',
+      isActive: true,
+    });
+    const categoryResponse = await axios.post(graphqlUrl, {
+      query: getCategoryData,
+      variables: {},
+    });
+    const result = categoryResponse?.data?.data?.categories;
+    assert.isNotEmpty(result);
+    categoryResponse.data.data.categories.forEach(item => {
```

Comment:
I think you should remove this loop one, because if another test adds a mainCategory it will fail.
and just validate the mainCategory with the one you created
```
    assert.equal(result[result.length - 1].mainCategory.title, mainCategory.title);
```

Reply:
Also just to clarify, it's not that I think it's wrong, but it might be too general of test if it has a relation or if it has a title present, because the title could be empty. 
We better off encapsulating things to their own, so no side effects occur, and keeping things as explicit as posible.

Reply:
You are right, it's better to encapsulate this, thanks for your feedback, I did it